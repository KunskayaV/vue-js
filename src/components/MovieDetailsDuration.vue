<template>
  <div class="duration-container">
    <span date-testid="details-year">{{ year }}</span>
    <span date-testid="details-duration">{{ minutes }} min</span>
  </div>
</template>

<script setup lang="ts">
import { getDuration } from '@/helpers/getDuration'
import { getYearFromDate } from '@/helpers/getYearFromDate'
import { computed } from 'vue'

const props = defineProps<{ date: string; duration: string }>()

const year = computed(() => getYearFromDate(props.date))
const minutes = computed(() => getDuration(props.duration))
</script>

<style scoped>
.duration-container {
  display: flex;
  gap: 24px;
  padding-block: 24px;
  font-size: 1rem;
  line-height: 1.5rem;
  color: #f65261;
}
</style>
