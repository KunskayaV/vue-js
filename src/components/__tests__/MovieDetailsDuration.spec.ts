import { describe, it, expect } from 'vitest'

import { mount } from '@vue/test-utils'
import MovieDetailsDuration from '../MovieDetailsDuration.vue'
import { mockMovie } from '@/__mocks__'

describe('MovieDetailsDuration', () => {
  it('should match snapshot', () => {
    const wrapper = mount(MovieDetailsDuration, {
      props: { date: mockMovie.releaseDate, duration: mockMovie.duration }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render correct data', () => {
    const wrapper = mount(MovieDetailsDuration, {
      props: { date: '2023-10-12', duration: 'PT90M' }
    })

    expect(wrapper.find('[date-testid=details-year]').text()).toBe('2023')
    expect(wrapper.find('[date-testid=details-duration]').text()).toBe('90 min')
  })
})
