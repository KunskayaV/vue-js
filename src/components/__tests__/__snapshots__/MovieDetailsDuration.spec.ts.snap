// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MovieDetailsDuration > should match snapshot 1`] = `"<div data-v-b9b942cb=\\"\\" class=\\"duration-container\\"><span data-v-b9b942cb=\\"\\" date-testid=\\"details-year\\">1995</span><span data-v-b9b942cb=\\"\\" date-testid=\\"details-duration\\">142 min</span></div>"`;
